acceptance/detector_test.go

```diff
@@ -0,0 +1,196 @@
+			h.AssertNil(t, err)
+			h.AssertEq(t, buildpackGroup.Group[0].ID, "simple_buildpack")
+			h.AssertEq(t, buildpackGroup.Group[0].Version, "simple_buildpack_version")
+
+			// check plan.toml
+			tempPlanToml := filepath.Join(copyDir, "layers", "plan.toml")
+			var buildPlan lifecycle.BuildPlan
+			_, err = toml.DecodeFile(tempPlanToml, &buildPlan)
+			h.AssertNil(t, err)
+			h.AssertEq(t, buildPlan.Entries[0].Providers[0].ID, "simple_buildpack")
+			h.AssertEq(t, buildPlan.Entries[0].Providers[0].Version, "simple_buildpack_version")
+			h.AssertEq(t, buildPlan.Entries[0].Requires[0].Name, "some_requirement")
+			h.AssertEq(t, buildPlan.Entries[0].Requires[0].Metadata["some_metadata_key"], "some_metadata_val")
+			h.AssertEq(t, buildPlan.Entries[0].Requires[0].Metadata["version"], "some_version")
+		})
+	})
+
+	when("environment variables are provided for buildpack and app directories and for the output files", func() {
+		var copyDir, containerName string
+
+		it.Before(func() {
+			containerName = "test-container-" + h.RandString(10)
+			var err error
+			copyDir, err = ioutil.TempDir("", "test-docker-copy-")
+			h.AssertNil(t, err)
+		})
+
+		it.After(func() {
+			if h.DockerContainerExists(t, containerName) {
+				h.Run(t, exec.Command("docker", "rm", containerName))
+			}
+			os.RemoveAll(copyDir)
+		})
+
+		it("writes group.toml and plan.toml in the right location and with the right names", func() {
+			h.DockerRunAndCopy(t,
+				containerName,
+				copyDir,
+				detectImage,
+				"/layers",
+				h.WithFlags("--user", userID,
+					"--env", "CNB_ORDER_PATH=/cnb/orders/always_detect_order.toml",
+					"--env", "CNB_BUILDPACKS_DIR=/cnb/custom_buildpacks",
+					"--env", "CNB_APP_DIR=/custom_workspace",
+					"--env", "CNB_GROUP_PATH=./custom_group.toml",
+					"--env", "CNB_PLAN_PATH=./custom_plan.toml",
+					"--env", "CNB_PLATFORM_DIR=/custom_platform",
+				),
+				h.WithArgs("-log-level=debug"),
+			)
+
+			// check group.toml
+			tempGroupToml := filepath.Join(copyDir, "layers", "custom_group.toml")
+			var buildpackGroup lifecycle.BuildpackGroup
+			_, err := toml.DecodeFile(tempGroupToml, &buildpackGroup)
+			h.AssertNil(t, err)
+			h.AssertEq(t, buildpackGroup.Group[0].ID, "always_detect_buildpack")
+			h.AssertEq(t, buildpackGroup.Group[0].Version, "always_detect_buildpack_version")
+
+			// check plan.toml - should be empty since we're using always_detect_order.toml so there is no "actual plan"
+			tempPlanToml := filepath.Join(copyDir, "layers", "custom_plan.toml")
+			planContents, err := ioutil.ReadFile(tempPlanToml)
+			h.AssertNil(t, err)
+			h.AssertEq(t, len(planContents) == 0, true)
+
+			// check platform directory
+			logs := h.Run(t, exec.Command("docker", "logs", containerName))
+			expectedPlatformPath := "platform_path: /custom_platform"
+			expectedAppDir := "app_dir: /custom_workspace"
+			h.AssertStringContains(t, string(logs), expectedPlatformPath)
+			h.AssertStringContains(t, string(logs), expectedAppDir)
```

Comment:
```suggestion
			h.AssertStringContains(t, logs, expectedPlatformPath)
			h.AssertStringContains(t, logs, expectedAppDir)
```
`h.Run` returns a `string`.

Reply:
Oops, you're right. It was probably something else before :)
I'll change it.